Improve Happiness System Summary (#545)
ldtteam/MinecoloniesWiki: source/systems/happinessandsaturation.md

## Happiness System

> There is an **overall** colony happiness and an **individual** citizen happiness. Higher colony happiness increases the initial level and skills of new colonists. Higher citizen happiness increases the rate the colonist's gains Intelligence when studying at a [Library](../../source/buildings/library).

Reviewer: ```suggestion
There is an **overall** colony happiness and an **individual** citizen happiness. Higher colony happiness increases the initial level and skills of new colonists. Higher citizen happiness increases the rate colonists gain Intelligence when studying at a [Library](../../source/buildings/library).
```

Overall colony happiness is 1-10 (initially set to 5). Happiness depends on three basic factors: **security, housing, and saturation**.

> If a colonist's saturation is above 7 shanks, they will be more happy.

Reviewer: ```suggestion
If a colonist's saturation is above 7 shanks, they will be happier.
```

> If a colonist's saturation above 7 shanks, they will be more happy.

Reviewer: ```suggestion
If a colonist's saturation above is 7 shanks, they will be happier.
```

> If the colonist's home (or for [guards](../../source/workers/guard) or trainees, workplace) is above level 2, the colonist will be more happy.

Reviewer: ```suggestion
If the colonist's home (or for [Guards](../../source/workers/guard) or trainees, their workplace) is above level 2.5, the colonist will be happier.
```

> If the colonist's home (or for [guards](../../source/workers/guard), or [trainees](../../source/workers/archerintraining), workplace) is above level 2, the colonist will be more happy.

Reviewer: ```suggestion
If a colonist's home (or for [Guards](../../source/workers/guard) or trainees, their workplace) is above level 3, the colonist will be happier.
```
Isn't this level 3? And since there are also Knights in Training, maybe remove the link entirely?

Author: The mechanical rule is BuildingLevel / 2.5, so they should return blue happiness at 2.5 and green happiness at level 2.75 or 3.  The distinction mostly matters for the Barracks, since it's the only building that can have a partial building level.

> If at least two Guards are available for every three workers, the colonists will be happier.

Reviewer: ```suggestion
If at least two Guards exist for every three citizens, the colonists will be happier.
```

> If two [Guards](../../source/workers/guard) are available for every three Workers, the colonists will be more happy.

Reviewer: ```suggestion
If at least two Guards are available for every three workers, the colonists will be happier.
```
If you already linked to the guard page in the last sentence you don't need to link it again

> These basic factors benefit from partial completion. Citizens will be happier with 5 shanks of saturation than 1, with 1 Guard for every 4 Workers than 1 Guard for every 10, and enjoy a level 2 house even more than a level 1 house. Colonists also become more happy as these values increase further, such as from upgrading a house to level 4 or 5, or being completely full on saturation.

Reviewer: ```suggestion
These basic factors benefit from partial completion. Citizens will be happier with 5 shanks of saturation than 1, with 1 Guard for every 4 workers than 1 Guard for every 10, and enjoy a level 2 house more than a level 1 house. Colonists also become happier as these values increase further, such as from upgrading a house to level 4 or 5 or having completely full saturation.
```

> Each time a citizen, other than a guard, dies, other citizens need to mourn the next day, and will be less happy for the next three days.  All citizens become slightly less happy for the next day when injured.

Reviewer: ```suggestion
Each time a citizen (other than a Guard) dies, other citizens will mourn the next day and be less happy for the next three days. All citizens become slightly less happy for the next day when injured.
```

> Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Citizens (other than guards) unable to make it to their bed at night to sleep will become unhappy. Fellow colonists' livelihood also matters for happiness (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).

Reviewer: ```suggestion
Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Citizens (other than Guards) unable to make it to their bed at night to sleep will become unhappy. Fellow colonists' livelihood also matters for happiness (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).
```

> Every night, colonists will also become less happy if sick, homeless, unemployed, or with nothing to do at their jobs, and will become increasingly unhappy the longer the problem persists. Injury will reduce colonist happiness, as will guards who are unable to sleep. Colonists will also become less happy if fellow colonists become sick, starving, homeless, or unemployed.

Reviewer: ```suggestion
Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Injuries will reduce colonist happiness, as will Guards who are unable to sleep. Colonists will also become less happy if fellow colonists become sick, starving, homeless, or unemployed (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).
```
"Guards who are unable to sleep" don't they automatically sleep? Why would they not be sleeping?

Author: Ah, think I misunderstood the behavior on that one.  Looks like guards are always going to have slept-tonight fulfilled (or at least at 1.0), while non-guard citizens are affected by sleeping in beds.

> Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Injuries will reduce colonist happiness, as will Guards who are unable to sleep. Colonists will also become less happy if fellow colonists become sick, starving, homeless, or unemployed (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).

Reviewer: ```suggestion
Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Injuries will also reduce colonist happiness, as will sleeplessness. Fellow colonists' livelihood also matters for happiness (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).
```
Maybe this then?

Author: I think I've got injuries twice, so pulled it from there and left the one that's next to deaths.  I tried a different wording that emphasizes that it's specifically inability to get to a bed to sleep that causes the unhappiness :
`Citizens (other than guards) unable to make it to their bed at night to sleep will become unhappy. Fellow colonists' livelihood also matters for happiness (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).`

Does that look better?

> Surviving a raid without losing any colonists provides a colony-wide happiness boost.

Reviewer: ```suggestion
Surviving a raid without losing any colonists provides a colony-wide happiness boost.
```
Does this include Guards dying? If no -> "without losing any citizens"

Author: raidManager.onLostCitizen() fires for all citizens, both guards and non-guards, and that's what controls the raidswithoutdeath bonus.  It's only death happiness modifier and need to mourn that skip guard deaths.
